indexd/index/drivers/alchemy.py

```diff
@@ -432,14 +433,26 @@ def ids(
             if urls_metadata or negate_params:
                 query = query.distinct(IndexRecord.did)
 
-            query = query.order_by(IndexRecord.did)
+            if page is not None:
+                # when we're paginating, order by updated date so newly added stuff is
+                # at the end (reduce risk that a new records ends up in a page
+                # earlier on) and allows for some logic to check for newly added records
+                # (e.g. parallelly processing from beginning -> middle and ending -> middle
+                #       and as a final step, checking the "ending"+1 to see if there are
+                #       new records).
+                query = query.order_by(IndexRecord.updated_date)
+            else:
+                query = query.order_by(IndexRecord.did)
```

Comment:
should we just always order by date?

Reply:
I thought about that... probably? I was afraid to change the existing behavior tbh, but I guess it shouldn't matter, you'll always get the next `did` regardless of the ordering